modules/http-client/src/main/java/org/jpos/http/client/HttpQuery.java

```diff
@@ -261,9 +263,27 @@ else if ("lax".equals(redirProp))
             redirectStrategy= LaxRedirectStrategy.INSTANCE;
         else
             throw new ConfigurationException("'redirect-strategy' must be 'lax' or 'default'");
+
+
+        String maxConn = System.getProperty("http.maxConnections");
+        if (maxConn == null || maxConn.trim().isEmpty()) {
+            String maxConnCfg = cfg.get("maxConnections", DEFAULT_MAX_CONNECTIONS);
```

Comment:
You can use `cfg.getInt` here, and change `DEFAULT_MAX_CONNECTIONS` from `String` to `int`

Reply:
Yes. The logic path was different initially, and the values needed to be stringified (because properties only work as strings). 

In the current code it could be done as you say, but:
1) I need to convert it back to String anyway, in order to call `System.setProperty()`.
2) I'm not doing math with those values, so they don't really need to be numbers at any time.
3) I'm having a final `parseInt` verification, after the `if`,  to see if the property is really a number, and that applies even if the logic didn't get inside the `if`.    
That covers the case when an **external** sys property was set and it didn't conform to a number .
Example that happened to me during testing, I had this java invocation:   
   ```java
   java -Dhttp.maxConnections=$HTTP_MAX_CONN  .... 
   ```   
   But at some point I forgot to actually give a value to `HTTP_MAX_CONN`, so the property **existed** but was read as an **empty string**, and it was giving me a weird exception later (an exception happening well inside Apache's code).  

So I thought about giving an early fail with a `ConfigurationException` that covers both cases of external and internal property, and there's no point in doing the earlier conversion to `int` inside the `if`.

Sorry for the long explanation. 
If you still want me to change it to `cfg.getIn`, I'll do it :-)